rosidl_generator_py/resource/_msg_support.c.em

```diff
@@ -61,10 +61,18 @@ full_classname = '%s.%s._%s.%s' % (spec.base_type.pkg_name, subfolder, module_na
 }@
   char full_classname_dest[@(len(full_classname) + 1)];
 
-  char * class_name = (char *)PyUnicode_1BYTE_DATA(
-    PyObject_GetAttrString(PyObject_GetAttrString(_pymsg, "__class__"), "__name__"));
-  char * module_name = (char *)PyUnicode_1BYTE_DATA(
-    PyObject_GetAttrString(PyObject_GetAttrString(_pymsg, "__class__"), "__module__"));
+  char * class_name = NULL;
+  char * module_name = NULL;
+  {
+    PyObject * attr1 = PyObject_GetAttrString(_pymsg, "__class__");
+    PyObject * attr2 = PyObject_GetAttrString(attr1, "__name__");
+    class_name = (char *)PyUnicode_1BYTE_DATA(attr2);
+    Py_DECREF(attr2);
+    attr2 = PyObject_GetAttrString(attr1, "__module__");
+    module_name = (char *)PyUnicode_1BYTE_DATA(attr2);
+    Py_DECREF(attr2);
+    Py_DECREF(attr1);
+  }
```

Comment:
Question: why is the new logic scoped everywhere in this PR ?

Reply:
That makes the scope of these temporary variables as small as possible. It makes it clear that they are not used beyond these few lines.